### PR TITLE
[FRONTEND][MXNET] Fix L2normalization error when run ssd model

### PR DESCRIPTION
**Pull request by titikid**
This PR fix error L2Normalization when run ssd models from MXNET frontend. I also faced another error when run it with target openCL as described here [https://github.com/dmlc/tvm/issues/1302](url). For short term solution, i reduced num_thread_x = 8 (line 307) and ofactor = 4 (line 309) in topi/python/topi/cuda/conv2d_nchw.py and i could run my ssd model successfully. How about long term solution for this issue?

### CODE REVIEW

File: nnvm/python/nnvm/frontend/mxnet.py [hunk 1]
```diff
@@ -263,6 +263,12 @@ def _expand_dims(inputs, attrs):
     new_attrs['axis'] = _required_attr(attrs, 'axis')
     return _get_nnvm_op(op_name)(*inputs, **new_attrs)
 
+def _l2_normalize(inputs, attrs):
+    op_name, new_attrs = 'l2_normalize', {}
+    mode = attrs.get('mode', 'instance')
+    new_attrs['axis'] = 0 if mode == 'instance' else 1
```
Comment (PariksheetPinjari909) [inline nnvm/python/nnvm/frontend/mxnet.py]: L2 normalize also support for spatial axis. Can consider 'spatial' case also

**Comment by PariksheetPinjari909** [comment]
Please add test case for new code added.

**Comment by kaishijeng** [comment]
Will this help mobilenet-ssd model?

Thanks,

**Comment by kaishijeng** [comment]
@kevinthesun

**Comment by kevinthesun** [comment]
@kaishijeng Yes. L2Normlization is required for gluoncv sad implementation: https://github.com/dmlc/gluon-cv/blob/master/gluoncv/model_zoo/ssd/ssd.py I plan to deploy these models in the next few weeks.

**Comment by kaishijeng** [comment]
@kevinthesun 

Do you plan to support OpenCL for mobilenet ssd?

Thanks,

**Comment by yzhliu** [comment]
Please rebase according to https://github.com/dmlc/tvm/issues/1394
Sorry for the inconvience

**Comment by kevinthesun** [comment]
@kaishijeng I think @Laurawly is working for OpenCL SSD support.

**Comment by yzhliu** [comment]
@titikid Could you update the status? wondering whether @Laurawly 's recent work fixes problems https://github.com/dmlc/tvm/pull/1578

File: nnvm/python/nnvm/frontend/mxnet.py (hunk 1 above)
Comment (titikid) [reply nnvm/python/nnvm/frontend/mxnet.py]: @PariksheetPinjari909  Ok. i already fixed this

**Comment by titikid** [comment]
@yzhliu i updated l2_normalize's implementation with test cases. Plz help me review it. Thanks!

**Comment by yzhliu** [comment]
@titikid need to rebase according to https://github.com/dmlc/tvm/issues/1394 . currently the PR has too many unrelated stuffs to review. Sorry for the inconvenience.

**Comment by titikid** [comment]
@yzhliu sorry. I got some issues when rebasing. May i make another PR?

**Comment by kaishijeng** [comment]
@kevinthesun 

Any progress of gluoncv ssd implementation: ?

Thanks,

**Comment by kevinthesun** [comment]
gluoncv ssd will be supported through hybrid frontend. It is delayed due to some limitation of hybrid frontend.

**Comment by madhavajay** [comment]
@kevinthesun Anything that we can help with?

**Comment by kaishijeng** [comment]
@kevinthesun 

   Any progress on this?

Thanks,

**Comment by kaishijeng** [comment]
@titikid @kevinthesun 

Does current TVM code support mobilenet-ssd yet?

Thanks,

**Comment by kevinthesun** [comment]
Have you tried mxnet_ssd? I think that should be supported now. Gluoncv models are in progress.

**Comment by kaishijeng** [comment]
Does it have mobilenet_ssd?
Thanks,

**Comment by kevinthesun** [comment]
https://github.com/zhreshold/mxnet-ssd contains mobilenet ssd.

**Comment by kaishijeng** [comment]
I am not able to run deploy_ssd.py successfully. So I don't think I can try mobilenet_ssd using deploy_ssd.py

Thanks,

**Comment by kevinthesun** [comment]
You can file a issue on discuss forum.

**Pull request closed without merging**